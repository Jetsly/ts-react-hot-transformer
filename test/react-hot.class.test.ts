import * as ts from 'typescript';
import { join } from 'path';
import { readdirSync, readFileSync, statSync } from 'fs';
import transformerFactory from '../src/react-hot.dev';
import transformerProdFactory from '../src/react-hot.prod';
const printer = ts.createPrinter();
const FIXTURES_DIR = join(__dirname, '__fixtures__');
function addRHLPlugin(prod = false) {
  return prod ? transformerProdFactory : transformerFactory;
}

describe('copies arrow function body block onto hidden class methods', () => {
  const fixturesDir = join(FIXTURES_DIR, 'class-properties');
  readdirSync(fixturesDir).forEach(fixtureName => {
    const fixtureFile = join(fixturesDir, fixtureName);
    const testFile: Array<string> = ['arguments.js'];
    // if (statSync(fixtureFile).isFile() && testFile.indexOf(fixtureName) > -1) {
    if (statSync(fixtureFile).isFile()) {
      it(fixtureName.split('-').join(' '), () => {
        const sourceCode = readFileSync(fixtureFile, 'utf-8');
        const source = ts.createSourceFile(fixtureName, sourceCode, ts.ScriptTarget.ES2016, true);
        const result = ts.transform(source, [addRHLPlugin()]);
        const transformedSourceFile = result.transformed[0];
        const resultCode = printer.printFile(transformedSourceFile);
        expect(resultCode).toMatchSnapshot();
      });
    }
  });
});

