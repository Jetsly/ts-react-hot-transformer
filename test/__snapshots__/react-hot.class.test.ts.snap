// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`copies arrow function body block onto hidden class methods arguments.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    bar = (a, b) => {
        arguments;
        return a(b);
    };
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"arguments.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods arrow function in constructor.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    constructor() {
        this.onClick = (e) => e.target.value;
    }
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"arrow-function-in-constructor.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods async functions expression body.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    bar = async (a, b) => await b(a);
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"async-functions-expression-body.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods async functions.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    bar = async (a, b) => {
        return await a(b);
    };
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"async-functions.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods block body.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    bar = (a, b) => {
        return a(b);
    };
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"block-body.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods default params.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    bar = (a = \\"foo\\") => {
        return \`\${a}bar\`;
    };
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"default-params.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods destructured params.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    bar = ({ a }, { b }) => {
        return \`\${a}\${b}\`;
    };
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"destructured-params.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods export default.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
const _default = {
    'app.title': 'BMS',
    'app.desc': 'A Manager platform for BMS',
};
export default _default;


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(_default, \\"default\\", \\"export-default.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods expression body.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    onClick = (e) => e.target.value;
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"expression-body.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods nested arguments.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    bar = (a, b) => {
        () => {
            arguments;
        };
        return a(b);
    };
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"nested-arguments.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods nested new.target.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    bar = (a, b) => {
        () => {
            new.target;
        };
        return a(b);
    };
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"nested-new.target.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods new.target.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    bar = (a, b) => {
        new.target;
        return a(b);
    };
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"new.target.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods not a function.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    bar = 2;
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"not-a-function.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods not an arrow function.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    bar = function (a, b) {
        return a(b);
    };
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"not-an-arrow-function.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods same name as class method.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    bar = (a, b) => {
        return a(b);
    };
    bar() {
        return 2;
    }
    public __reactstandin__regenerateByEval(key, code) { this[key] = eval(code); }
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"same-name-as-class-method.js\\");

leaveModule(module);
}());


;
"
`;

exports[`copies arrow function body block onto hidden class methods static property.js 1`] = `
"(function () {
     var enterModule = require('react-hot-loader').enterModule;
     enterModule && enterModule(module);
   }());


;
class Foo {
    static bar = (a, b) => {
        return a(b);
    };
}


;
(function () {
var reactHotLoader = require('react-hot-loader').default;
var leaveModule = require('react-hot-loader').leaveModule;

if (!reactHotLoader) {
  return;
}

reactHotLoader.register(Foo, \\"Foo\\", \\"static-property.js\\");

leaveModule(module);
}());


;
"
`;
